cardano-api/src/Cardano/Api/MetaData.hs

```diff
@@ -23,13 +26,103 @@ import qualified Data.Map.Strict as Map
 import qualified Data.Scientific as Scientific
 import qualified Data.Text as Text
 import qualified Data.Text.Encoding as Text
-
 import qualified Data.Vector as Vector
 
 import           Shelley.Spec.Ledger.MetaData (MetaData (..), MetaDatum (..))
 
 
 
+-- | A transaction metadata validation error.
+data TxMetadataValidationError
+  = TxMetadataTextStringInvalidLengthError
+    -- ^ The length of a text string metadatum value exceeds the maximum.
+      !Int
+      -- ^ Maximum byte length.
+      !Int
+      -- ^ Actual byte length.
+  | TxMetadataByteStringInvalidLengthError
+    -- ^ The length of a byte string metadatum value exceeds the maximum.
+      !Int
+      -- ^ Maximum byte length.
+      !Int
+      -- ^ Actual byte length.
+  deriving (Eq, Show)
+
+-- | The maximum byte length of a transaction metadata text string value.
+txMetadataTextStringMaxLength :: Int
+txMetadataTextStringMaxLength = 64
```

Comment:
Should call it `txMetadataTextStringMaxByteLength` because string length and byte length are different.  Also is this something we ought to export so that users of the code can use it as a constant?